src/v2/models/AppState.js

```diff
@@ -34,7 +34,8 @@ export default Model.extend({
     remediations: 'array',
     dynamicRefreshInterval: 'number',
     deviceFingerprint: 'string',
-    hooks: 'object' // instance of models/Hooks
+    hooks: 'object', // instance of models/Hooks
+    hasRendered: ['boolean', false, false],
```

Comment:
maintaining the accuracy of `hasRendered` would be harder in future, because developer needs to have this knowledge about when to set and when to unset it.. Instead i would devise some other way, probably triggering an event instead ?

Reply:
The only other solution I can think of is making it a property of the `BaseLoginRouter` class. That would make it harder to override, but I didn't like the idea of moving something that smells like "state" outside of `AppState`